### PR TITLE
Diesel Gen consumes fuel with 10x precision (#4917)

### PR DESCRIPTION
**Pull request by voidsong-dragonfly**
- Fuel is consumed every 10t
 - Fuel is consumed for all 10t at once
 - Datapack recipe format is unchanged and so needs no updates

### CODE REVIEW

File: src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java [hunk 1]
```diff
@@ -158,32 +161,33 @@ public void tickServer()
...
+						if(!active) {
+							active = true;
+							animation_fanFadeIn = 80;
+						}
+						tanks[0].drain(fluidConsumed, FluidAction.EXECUTE);
+						consumeTick = 10;
+					} else if(active) {
+						active = false;
+						animation_fanFadeOut = 80;
 					}
```
Comment (BluSunrize) [inline src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java]: Code style please.
Opening and closing brackets go in separate lines.

File: src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java [hunk 2]
```diff
@@ -158,32 +161,33 @@ public void tickServer()
...
-						.map(CapabilityReference::getNullable)
-						.filter(Objects::nonNull)
-						.collect(Collectors.toList());
-				if(!presentOutputs.isEmpty()&&
-						tanks[0].getFluidAmount() >= fluidConsumed&&
-						// Sort receivers by lowest input
-						EnergyHelper.distributeFlux(presentOutputs, output, false) < output)
+				consumeTick--;
+			}
+			if (consumeTick == 0) //Consume 10*tamount every 10t to allow for 1/10th mB amounts
```
Comment (BluSunrize) [inline src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java]: Typo in the comment? "tamount"

**Review by BluSunrize — CHANGES_REQUESTED** [review]
Only minor changes required, but:

I'm missing the reason for this PR. Can you illustrate why this change is necessary?

File: src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java (hunk 1 above)
Comment (voidsong-dragonfly) [reply src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java]: Hot damn I thought I fixed all those >.<

File: src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java (hunk 2 above)
Comment (voidsong-dragonfly) [reply src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java]: Ohhh - no, that's an abbreviation for "tick-amount"
Basically, "consume ten times the amount per tick every ten ticks"
Will explain better though

**Comment by voidsong-dragonfly** [comment]
Basically, we had talked about it on the discord, but fuel can only be either 25% better or 25% worse than biodiesel right now - which is a _lot_. Changing burn time by a whole mB is not a level of gradation that is particularly useful to work with unless you want enormous changes. This makes it so the fuel consumption system is not incredibly unwieldy to work with if you want finer gradations, allowing up to 2.5% gradation instead of 25%. I'll find the conversation in discord & ping you when I do

File: src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java [hunk 3]
```diff
@@ -158,32 +161,37 @@ public void tickServer()
...
-				List<IEnergyStorage> presentOutputs = outputs.stream()
-						.map(CapabilityReference::getNullable)
-						.filter(Objects::nonNull)
-						.collect(Collectors.toList());
-				if(!presentOutputs.isEmpty()&&
-						tanks[0].getFluidAmount() >= fluidConsumed&&
-						// Sort receivers by lowest input
-						EnergyHelper.distributeFlux(presentOutputs, output, false) < output)
+				consumeTick--;
+			}
```
Comment (BluSunrize) [inline src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java:172]: For single-line conditionals we usually skip the braces, but I can ignore that =p

**Review by BluSunrize — APPROVED** [review]
LGTM

File: src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java (hunk 3 above)
Comment (voidsong-dragonfly) [reply src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java:172]: I was making absolutely sure I followed braces codestyle after the issues I'd had :p

**Pull request merged**